pkg/autoscaler/metrics/collector.go

```diff
@@ -279,6 +279,10 @@ func newCollection(metric *av1alpha1.Metric, scraper StatsScraper, tickFactory f
 				scrapeTicker.Stop()
 				return
 			case <-scrapeTicker.C:
+				if c.metric.Spec.ScrapeTarget == "" {
```

Comment:
This needs to be based on `c.currentMetric()` to always get the latest state. Please store the result of `c.currentMetric()` in a variable first as it involves locking to get it.

Reply:
Fixed in 7b1feced1c20f12f662502628344d8087c3978f6.